specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/DataCollectorLogs.json

```diff
@@ -0,0 +1,221 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Log Analytics",
+    "description": "Azure Log Analytics API reference",
+    "version": "2020-03-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/dataCollectorLogs": {
```

Comment:
From description it seems like this POST action is used to delete a log table. what is the need to expose deletion operation through the control plane on something (tables) which is not exposed on the control plane. Its wierd and can cause of sorts of issues like permission check, inconsistent state/confusion for customer where i am assuming same thing can be from the dataplane api as well etc.

Reply:
and if we agree that this is the only way to go about it and its required, i will ask you, at the very least to change the name of this action. This POST action from the name seems very harmless but it ends up deleting something.  So name should indicate what this operation does. may be call it deleteDataCollectorLogsTable. But lets answer the above question first.

Reply:
it all derives from the fact this is not a typical CRUD, so post is multi purpose, in this case being used for delete.
i have no problem changing the post to delete, but there will be no put. is that sounds reasonable?

Reply:
yes, better to switch to GET and DELETE